### ISSUE TITLE
nacos 2.2.3  Caused by: java.util.concurrent.RejectedExecutionException:

### ISSUE THREAD

**Issue by wuqingfang** [alibaba/nacos#11717]
**Describe the bug**
The production environment nacos2.2.3+dubbo3.2.10 fails to connect to nacos after the application restarts after running for a period of time.

**Expected behavior**
The application can be accessed normally without restarting, and nacos cannot be connected after restarting the application.
error:
Caused by: java.lang.RuntimeException: publish nacos metadata failed
cause: publish nacos metadata failed, dubbo version: 3.2.10, current host: 29.4.232.51, error code: 1-37. This may be caused by , go to https://dubbo.apache.org/faq/1/37 to find instructions.
java.lang.RuntimeException: publish nacos metadata failed

**Actually behavior**
The nacos.log file keeps printing the following error message:
java.util.concurrent.RejectedExecutionException: Task io.grpc.internal.SerializingExecutor@12819f3d rejected from java.util.concurrent.ThreadPoolExecutor@76ae056f[Running, pool size = 512, active threads = 0, queued tasks = 16384, completed tasks = 19646742]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
        at io.grpc.internal.SerializingExecutor.schedule(SerializingExecutor.java:102)

netstat -anlp|grep 9848 |grep CLOSE_WAIT |wc -l
YN_RECV         91
CLOSE_WAIT       40961
ESTABLISHED      3
FIN_WAIT1        17
SYN_SENT         4

**How to Reproduce**
1. nacos 2.2.3 , startup.sh -m standalone
2. dubbo 3.2.10
3. nacos runs for half a day and then reappears
4. See errors

**Desktop (please complete the following information):**
 - OS: CentOS Linux release 7.0.1406 (Core)
 - Version [e.g. nacos-server 2.2.3, nacos-client 2.0.4、2.2.4]
 - Module [e.g. naming/config] default
 - SDK [dubbo3.2.10,]

**Additional context**

**Comment by KomachiSion** [comment]
java.util.concurrent.ThreadPoolExecutor@76ae056f[Running, pool size = 512, active threads = 0, queued tasks = 16384, completed tasks = 19646742]

the thread tasks queue is full and no new worker to handle new task, reject is right.

you can do some research to found out which tasks in your deploy env call frequently so that the thread pool full.

**Comment by wuqingfang** [comment]
But the thread here is the nacos internal thread, not the business request thread, as follows:

"nacos-grpc-executor-132" #440 daemon prio=5 os_prio=0 tid=0x00007ff82020c800 nid=0xe1b waiting on condition [0x00007ff77abea000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000862a8c20> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

   Locked ownable synchronizers:
	- None

**Comment by wuqingfang** [comment]
[nacos.log](https://github.com/alibaba/nacos/files/14149493/nacos.log)
[nacos-pid-2445.zip](https://github.com/alibaba/nacos/files/14149498/nacos-pid-2445.zip)

**Comment by wuqingfang** [comment]
> java.util.concurrent.ThreadPoolExecutor@76ae056f[Running, pool size = 512, active threads = 0, queued tasks = 16384, completed tasks = 19646742]
> 
> the thread tasks queue is full and no new worker to handle new task, reject is right.
> 
> you can do some research to found out which tasks in your deploy env call frequently so that the thread pool full.

How do we check

**Comment by KomachiSion** [comment]
> But the thread here is the nacos internal thread, not the business request thread, as follows:
> 
> "nacos-grpc-executor-132" #440 daemon prio=5 os_prio=0 tid=0x00007ff82020c800 nid=0xe1b waiting on condition [0x00007ff77abea000] java.lang.Thread.State: WAITING (parking) at sun.misc.Unsafe.park(Native Method) - parking to wait for <0x00000000862a8c20> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject) at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175) at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039) at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442) at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:748)
> 
> Locked ownable synchronizers: - None

which port pool? there are two pools for 9848 and 9849. maybe you see the wating pool is 9849 port but other pool is blocked for 9848.

**Comment by KomachiSion** [comment]
> > java.util.concurrent.ThreadPoolExecutor@76ae056f[Running, pool size = 512, active threads = 0, queued tasks = 16384, completed tasks = 19646742]
> > the thread tasks queue is full and no new worker to handle new task, reject is right.
> > you can do some research to found out which tasks in your deploy env call frequently so that the thread pool full.
> 
> How do we check

dump the memory and analyze by yourself or get the queue item by arthas or other java tools.

**Comment by wuqingfang** [comment]
[nacos.log](https://github.com/alibaba/nacos/files/14149493/nacos.log)
[nacos-pid-2445.zip](https://github.com/alibaba/nacos/files/14149498/nacos-pid-2445.zip)

Don't see anything wrong，the dump file is large.
[2445_Leak_Suspects.zip](https://github.com/alibaba/nacos/files/14188117/2445_Leak_Suspects.zip)

**Comment by KomachiSion** [comment]
Please get the `ThreadPoolExecutor@76ae056f` by dump file and analyze which task in the queued and search the workers block which part.
And print the key information to issue comment.
Or anyone in community prefer to help to analyze the whole dump file in upload?

**Comment by wuqingfang** [comment]
> > But the thread here is the nacos internal thread, not the business request thread, as follows:
> > "nacos-grpc-executor-132" #440 daemon prio=5 os_prio=0 tid=0x00007ff82020c800 nid=0xe1b waiting on condition [0x00007ff77abea000] java.lang.Thread.State: WAITING (parking) at sun.misc.Unsafe.park(Native Method) - parking to wait for <0x00000000862a8c20> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject) at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175) at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039) at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442) at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:748)
> > Locked ownable synchronizers: - None
> 
> which port pool? there are two pools for 9848 and 9849. maybe you see the wating pool is 9849 port but other pool is blocked for 9848.
肯定是9848
